client/app/shared/ci-helper/ci-helper.controller.js

```diff
@@ -1054,47 +1134,49 @@ const CiHelperController = function CiHelperController(
     };
 
     function initController() {
-        clearLauncher();
+        resetLauncher(); //TODO: why do we need this?
         const launchersPromise = getAllLaunchers()
-            .then(function (launchers) {
-                return $scope.launchers = launchers;
+            .then(launchers => {
+                return vm.launchers = launchers;
             });
-        toolsService.fetchIntegrationOfTypeByName('AUTOMATION_SERVER').then((res) => {
-            if (res.success) {
-                $scope.servers = res.data;
-                if($scope.servers.length > 1) {
-                    $scope.needServer = true;
+        toolsService.fetchIntegrationOfTypeByName('AUTOMATION_SERVER')
+            .then(response => {
+                if (response.success) {
+                    $scope.servers = response.data || [];
+                    $scope.needServer = $scope.servers.length > 1;
                 }
-            }
-        });
-        getTenantInfo().then(function (tenant) {
-            isMultitenant = tenant.multitenant;
-            getClientId()
-                .then(clientId => {
-                    $scope.clientId = clientId;
-                });
-        });
-        const scmAccountsPromise = ScmService.getAllScmAccounts().then(function (rs) {
-            return $q(function (resolve, reject) {
-                if (rs.success) {
-                    if (rs.data && rs.data.length) {
-                        $scope.scmAccounts = rs.data.filter(function (scmAccount) {
-                            return scmAccount.repositoryURL;
-                        });
+            });
+        getTenantInfo()
+            .then(tenant => {
+                isMultitenant = tenant.multitenant;
+                getClientId()
+                    .then(clientId => {
```

Comment:
You can use the following way to remove nesting:
```
getTenantInfo()
  .then(tenant => {
    isMultitenant = tenant.multitenant;
    return getClientId();
  })
  .then(clientId => $scope.clientId = clientId);
```

Reply:
done